Neos.Neos/Classes/Routing/FrontendNodeRoutePartHandler.php

```diff
@@ -75,6 +86,17 @@ class FrontendNodeRoutePartHandler extends DynamicRoutePart implements FrontendN
      */
     protected $contentDimensionPresetSource;
 
+    /**
+     * @Flow\Inject
+     * @var NodeShortcutResolver
+     */
+    protected $nodeShortcutResolver;
+
+    /**
+     * @var Site[] indexed by the corresponding host name
+     */
+    protected $siteByHostRuntimeCache = [];
```

Comment:
This handler is a prototype (and it has to be since it is stateful) but the routing framework will re-use the same instance for the same route.
So 100 FE URLs on one page would previously resolve the current site 100 times

Reply:
Yep, cool, that makes sense